test/Cruzo1155Vault.ts

```diff
@@ -0,0 +1,300 @@
+        collectionURI: "https://cruzo.io/collection",
+    };
+
+    beforeEach(async () => {
+        [owner, seller, buyer, claimer, claimer2] = await ethers.getSigners();
+
+        const CruzoMarket = await ethers.getContractFactory("CruzoMarket");
+        const Cruzo1155 = await ethers.getContractFactory("Cruzo1155");
+        const Factory = await ethers.getContractFactory("Cruzo1155Factory");
+        const Vault = await ethers.getContractFactory("Cruzo1155Vault");
+
+        market = await upgrades.deployProxy(CruzoMarket, [serviceFee, rawVaultFuncSignature], {
+            kind: "uups",
+        });
+        await market.deployed();
+        beacon = await upgrades.deployBeacon(Cruzo1155);
+        await beacon.deployed();
+        factory = await Factory.deploy(
+            beacon.address,
+            RAW_FACTORY_INITIALIZE_SIGNATURE,
+            "https://cruzo.market",
+            market.address
+        );
+        await factory.deployed();
+        vault = await upgrades.deployProxy(Vault, [market.address], {
+            kind: "uups",
+        });
+        await vault.deployed();
+        await market.setVaultAddress(vault.address);
+    });
+
+    describe("Simple token gift without wallet", () => {
+        it("Give and claim a gift", async () => {
+            const tokenId = ethers.BigNumber.from("1");
+            const supply = ethers.BigNumber.from("100");
+            const tradeAmount = ethers.BigNumber.from("10");
+            const price = ethers.utils.parseEther("0.01");
+            const utf8Encode = new TextEncoder();
+            const secretKey = "skdjf87ddg9ddfud9d9fu"
+            const hash = keccak256(utf8Encode.encode(secretKey))
+
+            const createTokenTx = await factory
+                .connect(owner)
+                .create(
+                    tokenDetails.name,
+                    tokenDetails.symbol,
+                    tokenDetails.collectionURI
+                );
+            const createTokenReceipt = await createTokenTx.wait();
+            const createTokenEvent = getEvent(createTokenReceipt, "NewTokenCreated");
+            token = await ethers.getContractAt(
+                "Cruzo1155",
+                createTokenEvent.args?.tokenAddress
+            );
+            expect(
+                await token
+                    .connect(seller)
+                    .create(tokenId, supply, seller.address, "", [])
+            );
+            await expect(
+                market
+                    .connect(seller)
+                    .openTrade(token.address, tokenId, tradeAmount, price)
+            )
+                .emit(market, "TradeOpened")
+                .withArgs(token.address, tokenId, seller.address, tradeAmount, price);
+
+            expect(await market.connect(buyer).giftItemViaVault(token.address, tokenId, seller.address, tradeAmount, hash, { value: tradeAmount.mul(price) }))
+            expect(await vault.connect(claimer).claimGiftForMyself(secretKey))
```

Comment:
You can also add an intermediate check that Vault contract has tokens on its balance after "gift" and before "claim". And then check that it has 0 tokens in the end (after "claim").